src/charm.py

```diff
@@ -1722,6 +1724,11 @@ def _handle_processes_failures(self) -> bool:
         # Restart the PostgreSQL process if it was frozen (in that case, the Patroni
         # process is running by the PostgreSQL process not).
         if self._unit_ip in self.members_ips and self._patroni.member_inactive:
+            data_directory_contents = os.listdir(POSTGRESQL_DATA_PATH)
+            if len(data_directory_contents) == 1 and data_directory_contents[0] == "pg_wal":
+                os.remove(os.path.join(POSTGRESQL_DATA_PATH, "pg_wal"))
+                logger.info("PostgreSQL data directory was not empty. Removed pg_wal")
+                return True
```

Comment:
This was needed for async replication, when sometimes the pg_wal folder got recreated at some point with old data.